src/log.c

```diff
@@ -236,19 +164,19 @@ static int writeEntry(Log *log, raft_entry_t *ety)
 
 static int writeHeader(Log *log)
 {
-    ssize_t off = 0;
+    ssize_t n = 0;
     char buf[1024];
 
-    off += writeLength(buf + off, sizeof(buf) - off, '*', RAFTLOG_ELEM_COUNT);
-    off += writeString(buf + off, sizeof(buf) - off, RAFTLOG_STR);
-    off += writeLong(buf + off, sizeof(buf) - off, RAFTLOG_VERSION);
-    off += writeString(buf + off, sizeof(buf) - off, log->dbid);
-    off += writeLong(buf + off, sizeof(buf) - off, log->node_id);
-    off += writeLong(buf + off, sizeof(buf) - off, log->snapshot_last_term);
-    off += writeLong(buf + off, sizeof(buf) - off, log->snapshot_last_idx);
+    n += multibulkWriteLen(buf + n, sizeof(buf) - n, '*', RAFTLOG_ELEM_COUNT);
+    n += multibulkWriteStr(buf + n, sizeof(buf) - n, RAFTLOG_STR);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, RAFTLOG_VERSION);
+    n += multibulkWriteStr(buf + n, sizeof(buf) - n, log->dbid);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, log->node_id);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, log->snapshot_last_term);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, log->snapshot_last_idx);
 
     if (truncateFiles(log, 0, 0) != RR_OK ||
-        FileWrite(&log->file, buf, off) != off ||
+        FileWrite(&log->file, buf, n) != n ||
```

Comment:
I don't understand this change.  why did it change from `off` to 0?

Reply:
never mind n, not 0, I think `n` is a bit problematic here, `off` was better imo.